scripts/corr2_dsim_control.py

```diff
@@ -109,6 +134,71 @@
     IPython.embed()
     something_happened = True
 
+if args.set_sine_source:
+    """Sine source selection field, including source ,scale and frequency"""
+    sine_source = args.set_sine_source[0]
+    xscale = float(args.set_sine_source[1])
+    yfreq = float(args.set_sine_source[2])
+    try:
+        SineSource = getattr(dhost.sine_sources, 'sin_{}'.format(sine_source))
+        maxfreq = "{}MHz".format( SineSource.max_freq / 1e6 )
+        #print maxfreq
+        SineSource.set(scale = xscale, frequency = yfreq)
+        print "Sine sourced:", SineSource.name
+        print "Scale:", SineSource.scale
+        print "Frequency:", SineSource.frequency        
+    except AttributeError:
+        print "You can only select between, '0', '1'"
+        sys.exit(1)
+    except ValueError:
+        print "Error, verify your inputs"
+        sys.exit(1)
+    import IPython
+    IPython.embed()
+    something_happened = True
+
+if args.set_noise_source:
+    """Noise selection, selectiong source of noise and scale"""
+    NoiseSource = args.set_noise_source[0]
+    NoiseScale = float(args.set_noise_source[1])
+    ##maxfreq = "{}MHz".format( dhost.sine_sources.sin_0.max_freq / 1e6 )
+    try:
+        SourceFrom = getattr(dhost.noise_sources, 'noise_{}'.format(NoiseSource))
+        SourceFrom.set(scale = NoiseScale)
+        print "Noise Source:", SourceFrom.name
+        print "Noise Scale:", SourceFrom.scale
+    except AttributeError:
+        print "You can only select between, '0', '1', or 'corr'"
```

Comment:
It would be better if you could get the available sources from the dhost object rather than hardcoding it in the script.